crud/common/sharding_key.lua

```diff
@@ -115,6 +116,87 @@ function sharding_key_module.fetch_on_router(space_name)
     return cache.sharding_key_def
 end
 
+-- Make sure sharding key definition is a part of primary key.
+function sharding_key_module.is_part_of_pk(space_name, space_format,
+                                           primary_index, sharding_key_def)
+    dev_checks('string', 'table', 'table', 'table')
+
+    if cache.as_index_object[space_name] == nil then
+        cache.as_index_object[space_name] =
+            sharding_key_module.as_index_object(space_format, space_name,
+                                                sharding_key_def)
+    end
+    local sharding_key_as_index_obj = cache.as_index_object[space_name]
+
+    if cache.is_part_of_pk[space_name] ~= nil then
+        return cache.is_part_of_pk[space_name]
+    end
+
+    local is_part_of_pk = true
+    local pk_fieldno_map = utils.get_index_fieldno_map(primary_index.parts)
+    for _, part in ipairs(sharding_key_as_index_obj.parts) do
+        if pk_fieldno_map[part.fieldno] == nil then
+            is_part_of_pk = false
+            break
+        end
+    end
+    cache.is_part_of_pk[space_name] = is_part_of_pk
+
+    return is_part_of_pk
+end
+
+-- Build an array with sharding key values.
+function sharding_key_module.extract_from_index(sharding_key_def,
+                                                index_parts,
+                                                sharding_key_fieldno_map)
```

Comment:
The same as for `is_part_of_pk()`: let's lower it to local and export from `sharding_key.internal`.

Reply:
I don't understand this function. All arguments look as metainfo (no values). But the variable that we populate and returns is named as `sharding_key` and the comment above states 'Build an array with sharding key **values**'.

Reply:
As I see from the caller side, the 'sharding_key_def' argument is the primary key (its value, not definition).

Reply:
> The same as for is_part_of_pk(): let's lower it to local and export from sharding_key.internal.

Made it internal.

Reply:
> All arguments look as metainfo (no values). But the variable that we populate and returns is named as sharding_key and the comment above states 'Build an array with sharding key values'.

Comment is correct, function really returns an array with sharding key values.

Updated name of the first argument and comment.

```lua
--- a/crud/common/sharding_key.lua
+++ b/crud/common/sharding_key.lua
@@ -141,11 +141,13 @@ local function is_part_of_pk(space_name, space_format, primary_index, sharding_k
 end
 
 -- Build an array with sharding key values.
-local function extract_from_index(sharding_key_def, index_parts, sharding_key_fieldno_map)
+-- Function extracts those values from primary key that
+-- are a part of sharding key (passed as map with fieldno's).
+local function extract_from_index(primary_key, index_parts, sharding_key_fieldno_map)
     dev_checks('table', 'table', 'table')
 
     local sharding_key = {}
-    for i, k in ipairs(sharding_key_def) do
+    for i, k in ipairs(primary_key) do
         local fieldno = index_parts[i].fieldno
         if sharding_key_fieldno_map[fieldno] == true then
             table.insert(sharding_key, k)
```